### DISCUSSION TITLE
Push problem with console (maybe 2fa)

### DISCUSSION THREAD

**Discussion by FededelRincon** [community/community#22175] [New to GitHub]
<p>Im new with git, i dont know if it matthers, but im in new in linux also.</p>
<p>How i push until yesterday, i go my console, git push, put my Username and Password, and its done. no issue here.<br>
But, Today I recibe an email that says using password to access my repository will be deprecated, that i need to do a “Two-factor authentication”.</p>
<p>I did the Two-factor authentication (i have the 16 lines code), so now i want to  push from console like i use to, but…<br>
Username for “my”<br>
Password for “my”     (i checked the password)<br>
remote: Invalid username or password.<br>
fatal: Autenticación fails to ‘<a href="https://github.com/myName/myRepository.git/" rel="noopener nofollow ugc">https://github.com/myName/myRepository.git/</a>’</p>
<p>The question its how push to my repository now ?? (i dont want use visual studio code).</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>You need to <a href="https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/creating-a-personal-access-token">create a PAT (personal access token)</a> and use that instead of the password. You’ll need the <code>repo</code> scope, and possibly <code>workflow</code> if you use Github Actions (see <a href="https://docs.github.com/en/free-pro-team@latest/developers/apps/scopes-for-oauth-apps#available-scopes">Available scopes</a>).</p>
<p>Or alternatively you can <a href="https://docs.github.com/en/free-pro-team@latest/github/authenticating-to-github/connecting-to-github-with-ssh">use SSH instead of HTTPS</a>.</p>

**Comment by FededelRincon** [comment]
<p>thanks airtower-luna. The PAT thing works.</p>
<p>Maybe i need to make another question, but i give it a shot here…<br>
I dont want remember this 40 characters token (i believe that git f*** up hugh here).<br>
Is any chante to set it up account and token/password in my console and do “git push” and the console dont ask anything ??</p>
<p>I try this<br>
“git config --global credential.helper store”</p>
<p>and this using the token as a password<br>
“git config --global user.name “your username””<br>
“git config --global user.password “your password””<br>
But none work…</p>

**Comment by airtower-luna** [comment]
<p>The <code>user.*</code> settings have nothing to do with authentication, they are about author information in commits. And <code>user.password</code> isn’t a valid setting at all, so it won’t be used (and I strongly recommend you delete it immediately for security).</p>
<p>I can’t say much about the git credential helpers, I just use SSH. But in general I strongly advise against storing the token unencrypted. The best option is to use a password manager that stores your passwords in an encrypted database (I like <a href="https://www.keepassxc.org/">KeePassXC</a>), so you have to remember only one password, and of course keep the database backed up. The <code>cache</code> credential helper plus a password manager might be a good compromise.</p>

**Comment by FededelRincon** [comment]
<p>And I´m goingo to start using KeePassXC… It seems a good “solution”.<br>
And i will read more about SSH.</p>
<p>Thanks for share information, i also delete the user.* staffs</p>

**Comment by airtower-luna** [comment]
<p>Don’t delete <em>all</em> the <code>user.*</code> settings, those are important. Just the <code>user.password</code> one isn’t. :wink:</p>
